_posts/2024-07-03-oidc-cient-request.adoc

```diff
@@ -0,0 +1,440 @@
+
+=== Generating Key Pairs
+
+We can use the `keytool` tool in the cli to generate our keys and store them in a keystore file. We can generate keystore files of type `JKS` and `PKCS12` for our application. The general format for doing this is as follows:
+```
+    $ keytool -genkeypair -alias <keystore_alias> -keyalg <algorithm> -keysize <key_size> -validity <validity_in_days>
+-keystore <keystore_name> -dname "<distinguished_name>" -keypass <private_key_password> -storepass <keystore_password>
+```
+
+==== Generating PKCS12 Keystore
+
+Let us first create a PKCS12 keystore file with 2 different keys. Let's first generate an RSA keyPair: 
+```
+    $ keytool -genkeypair -alias rsaKey -keyalg RSA -keysize 2048 -validity 365 -keystore Keycloak.keystore.pkcs12 -dname "CN=client" -keypass password -storepass password
+```
+And for Elliptical Curve keys, use the following commands: 
+```
+    $ keytool -genkeypair -alias ecKey -keyalg EC -groupname secp256r1 -validity 365 -keystore Keycloak.keystore.pkcs12 -dname "CN=client" -keypass password -storepass password
+``` 
+Note that in this case, we have specified the *key size* for the RSA keys, and the *groupname* for Elliptical Curve keys.
+For the RSA keys, the same keysize can be used for RSA algorithms with different SHA values (i.e. RSA256, RSA512).
+However, this is not the case for Elliptical Curve keys. For ES256 algorithms, you need to use the groupname associated with the SHA value (i.e. `-groupname secp256r1` and for ES384, use `-groupname secp384r1` and so on). You can use the following commands to see the contents of the keystore: 
+```
+    $ keytool -list -v -keystore Keycloak.keystore.pkcs12 -storepass password
+```
+
+If you would like to use JKS keystore files instead, see https://docs.redhat.com/en/documentation/red_hat_jboss_data_virtualization/6.4/html/security_guide/create_a_privatepublic_key_pair_with_keytool[here] for more information.
+
+== Log Into the OpenShift Cluster
+Before we can deploy our application, we need to log into an OpenShift cluster. You can log in via the OpenShift CLI:
+```
+oc login -u myUserName
+```
+
+Alternatively, to log in using the API token, navigate to Developer Sandbox, click on your username at the top right corner, and click on Copy login command. It will redirect you to a page where you can copy your login command and paste it on your terminal. The command will be in this format:
+```
+oc login --token=myToken --server=myServerUrl
+```
+
+Once you have your environment set up with the required tools, we can move on to the next step to build and deploy our application on OpenShift.
+
+== Start Keycloak
+We will be using Keycloak as our OpenID identity provider.
+
+To start a Keycloak server in your project on OpenShift, use the following command:
+
+```
+oc process -f https://raw.githubusercontent.com/keycloak/keycloak-quickstarts/latest/openshift/keycloak.yaml \
```

Comment:
This cmd format did not work for me.   I had to remove the '\' and provide all text on a single line.
(oc process -f https://raw.githubusercontent.com/keycloak/keycloak-quickstarts/latest/openshift/keycloak.yaml  -p KEYCLOAK_ADMIN=admin  -p KEYCLOAK_ADMIN_PASSWORD=admin  -p NAMESPACE=<PROJECT_NAME>  oc create -f -  )

Reply:
right, I believe this is formatted like that to add the annotations <1>, <2> and so on. 
Some of the other guides also had the same formatting for this command. 
I have added `//` in front of the annotations to make that more clear.